### PR TITLE
Remove old "SFTP" implementation

### PR DESCRIPTION
**Pull request by NobodyXu**
This PR is split from #30 so that it can be merged in before release of v0.9.0 and avoid possible breaking changes required.

Signed-off-by: Jiahao XU <Jiahao_XU@outlook.com>

### CODE REVIEW

**Comment by jonhoo** [comment]
This change is [Reviewable](https://reviewable.io/reviews/openssh-rust/openssh/50)

File: src/changelog.rs [hunk 1]
```diff
@@ -9,6 +9,13 @@ use crate::*;
 /// ## Added
 ///  - Added changelog
 ///  - Added [`SessionBuilder::compression`]
+///  - Feature `scp`.
+///  - New mod [`scp`], contains the old SFTP implementation.
+///  - Added [`Session::scp`].
+///
+/// ## Removed
+///  - `Sftp`, `Mode` and `RemoteFile`.
```
Comment (jonhoo) [inline src/changelog.rs:17]: These weren't removed as far as I can tell?

File: Cargo.toml [hunk 2]
```diff
@@ -32,6 +32,7 @@ rustdoc-args = ["--cfg", "docsrs"]
 default = ["process-mux"]
 process-mux = []
 native-mux = ["openssh-mux-client"]
+scp = []
```
Comment (jonhoo) [inline Cargo.toml:35]: Hmm, why call this `scp`? That's already a separate command, but not one that this implementation uses as far as I can recall?

File: src/changelog.rs (hunk 1 above)
Comment (NobodyXu) [reply src/changelog.rs:17]: After mod `sftp` are renamed into `scp`, I removed these exports so now user have to type `scp::Sftp`.

Calling them removed is a little bit confusing, I will fix that.

File: Cargo.toml (hunk 2 above)
Comment (NobodyXu) [reply Cargo.toml:35]: Yeah, but I couldn't figure out an alternative name though.

File: src/changelog.rs (hunk 1 above)
Comment (NobodyXu) [reply src/changelog.rs]: Done.

File: Cargo.toml (hunk 2 above)
Comment (jonhoo) [reply Cargo.toml:35]: I think I'd go with something more descriptive, like `sh-file-opts`.

Alternatively we _could_ delete the module entirely. After all, There's nothing magical in there, and maybe it's better to shrink the crate's API and have users invoke `cat` and friends if they want to do file ops themselves..

Comment (NobodyXu) [reply Cargo.toml:35]: Well, if removing this module altogether is acceptable, then I will definitely remove it instead of keeping it around.

Comment (jonhoo) [reply Cargo.toml:35]: Yeah, let's do that :+1:

Comment (NobodyXu) [reply Cargo.toml]: Done!

File: tests/openssh.rs [hunk 3]
```diff
@@ -376,62 +374,63 @@ macro_rules! assert_remote_kind {
 
 #[tokio::test]
 #[cfg_attr(not(ci), ignore)]
-async fn sftp_can() {
+#[cfg(feature = "scp")]
```
Comment (jonhoo) [inline tests/openssh.rs:377]: These can go away now too!

**Review by jonhoo — CHANGES_REQUESTED** [review]

File: tests/openssh.rs (hunk 3 above)
Comment (NobodyXu) [reply tests/openssh.rs]: Thanks for reminding me!

The integration tests passed, so I forgot to remove them.

**Review by jonhoo — APPROVED** [review]

**Pull request merged**